### PR TITLE
Enable shell error checks

### LINKED ISSUE #4983: Enable error checks in linter.sh
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

We don't enable `set -o nounset`, `set -o errexit`, `set -o pipefail` in linter.sh, and this may lead to hiding errors.

### Expected Behavior

The shell should catch these errors.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ferrarimarco**
# Proposed changes

Enable error checks to:

- Exit on errors
- Disallow empty variables
- Fail when a piped command errors

Fixes #4983

## Readiness checklist

In order to have this pull request merged, complete the following tasks.

### Pull request author tasks

- [x] I included all the needed documentation for this change.
- [x] I provided the necessary tests.
- [x] I squashed all the commits into a single commit.
- [x] I followed the [Conventional Commit v1.0.0 spec](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I wrote the necessary upgrade instructions in the [upgrade guide](../docs/upgrade-guide.md).
- [x] If this pull request is about and existing issue,
  I added the `Fix #ISSUE_NUMBER` label to the description of the pull request.

### Super-linter maintainer tasks

- [x] Label as `breaking` if this change breaks compatibility with the previous released version.
- [x] Label as either: `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -200,8 +200,8 @@ You can configure super-linter using the following environment variables:
...
-| **FILTER_REGEX_INCLUDE**                        | `all`                           | Regular expression defining which files will be processed by linters (ex: `.*src/.*`)                                                                                                                                |
+| **FILTER_REGEX_EXCLUDE**                        | not set                         | Regular expression defining which files will be excluded from linting  (ex: `.*src/test.*`). Not setting this variable means to process all files.                                                                   |
+| **FILTER_REGEX_INCLUDE**                        | not set                         | Regular expression defining which files will be processed by linters (ex: `.*src/.*`). Not setting this variable means to process all files.                                                                         |
```
Comment (Hanse00) [inline README.md:204]: It might be worth explicitly calling out how these two variables interact as well.

File: lib/functions/setupSSH.sh [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 function SetupSshAgent() {
   # Check to see if a SSH_KEY_SECRET was passed
-  if [ -n "${SSH_KEY}" ]; then
+  if [ -n "${SSH_KEY:-}" ]; then
```
Comment (Hanse00) [inline lib/functions/setupSSH.sh:5]: Sorry if it's an obvious question, but I'm a bit rusty in the shell scripting department. What's the purpose of expanding the parameter with nothing?

File: README.md (hunk 1 above)
Comment (ferrarimarco) [reply README.md:204]: They don't have any interaction. We might want to call out which one we evaluate first?

Comment (Hanse00) [reply README.md:204]: Yeah, that's what I meant, interaction as in "What happens when I set both?", not necessarily that they interact directly on a programmatic level.

Comment (ferrarimarco) [reply README.md]: Done! PTAL :)

**Review by Hanse00 — APPROVED** [review]

**Pull request merged**